picard/ui/metadatabox/__init__.py

```diff
@@ -334,12 +322,14 @@ def _copy_value(self):
             # Build the mimedata to use for the clipboard
             mimedata = QtCore.QMimeData()
             converted_data_cache = {}
-            for mimetype, converters in self.mimedatas.items():
-                if converters.encode_func:
+            for result in self.mimedata_helper.encode_funcs():
+                mimetype = result[0]
+                encode_func = result[1]
```

Comment:
I'm not sure about returning a list like this from a yield - feels a little unpleasant

Reply:
```python
for mimetype, encode_func in self.mimedata_helper.encode_funcs():
     try:
```

This should work. And we don't need to `if encode_func:` because it is done in https://github.com/StevilKnevil/picard/pull/3/files#diff-94dd759005b2bce0911ed122946e3efa0584be49a6f84d8765934697c74d54ccR74


Also we don't need the `mimetype`, so we could just have:

```python

    def encode_funcs(self):
        """
        Return a generator of registered MIMETYPES and their handling encoding functions.
        """
        for mimetype, converters in self._registry.items():
            if converters.encode_func:
                yield self._registry[mimetype].encode_func
```


And:

```python
for encode_func in self.mimedata_helper.encode_funcs():
     try:
```

Reply:
Ah I missed that, sorry. Then just keep the generator but  `yield mimetype, self._registry[mimetype].encode_func`:

```python
    def encode_funcs(self):
        """
        Return a generator of registered MIMETYPES and their handling encoding functions.
        """
        for mimetype, converters in self._registry.items():
            if converters.encode_func:
                yield mimetype, self._registry[mimetype].encode_func
```

And:

```python
for mimetype, encode_func in self.mimedata_helper.encode_funcs():
     try:
```

Reply:
Oh yeah, I didn't think you could yield multiple values